kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/DerivedResourcesTest.java

```diff
@@ -155,6 +155,13 @@ public Class<R> resourceType() {
 
         @Override
         public Map<String, R> invokeDesired(P primary, Context<P> context) {
+
+            if (reconcilePrecondition != null) {
```

Comment:
I don't like this, but I think this is good enough for now.  
https://github.com/kroxylicious/kroxylicious/issues/2037